packages/webapp/src/routes/land/check-and-submit.js

```diff
@@ -34,22 +36,45 @@ const handlers = {
 }
 
 const getContext = request => {
+  const applicationDetails = application(request.yar, request.auth.credentials.account).landownerGainSiteRegistration
   return {
     listArray,
     boolToYesNo,
     dateToString,
     hideClass,
-    hideConsent: (request.yar.get(constants.redisKeys.ROLE_KEY) === 'Landowner' && request.yar.get(constants.redisKeys.LANDOWNERS)?.length === 0),
-    changeLandownersHref: request.yar.get(constants.redisKeys.ROLE_KEY) === 'Landowner' ? constants.routes.REGISTERED_LANDOWNER : constants.routes.ADD_LANDOWNERS,
+    application: applicationDetails,
+    hideConsent: (request.yar.get(constants.redisKeys.LANDOWNERS)?.length === 0),
+    changeLandownersHref: constants.routes.ADD_LANDOWNERS,
     routes: constants.routes,
     landownerNames: getAllLandowners(request.yar),
     legalAgreementType: request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE) &&
-      getLegalAgreementDocumentType(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE)),
-    legalAgreementParties: request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_PARTIES) && getNameAndRoles(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_PARTIES)),
+    getLegalAgreementDocumentType(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE)),
+    legalAgreementFileNames: getLegalAgreementFileNames(applicationDetails.files),
+    responsibleBodies: getResponsibleBodies(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_RESPONSIBLE_BODIES)),
+    landowners: getLandowners(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_LANDOWNER_CONSERVATION_CONVENANTS)),
+    habitatPlanSeperateDocumentYesNo: request.yar.get(constants.redisKeys.HABITAT_PLAN_LEGAL_AGREEMENT_DOCUMENT_INCLUDED_YES_NO),
+    getFileNameByType,
+    HabitatPlanFileName: getFileNameByType(applicationDetails.files, 'habitat-plan'),
+    localAndChargeFileName: getFileNameByType(applicationDetails.files, 'local-land-charge'),
+    HabitatWorksStartDate: getDateString(request.yar.get(constants.redisKeys.ENHANCEMENT_WORKS_START_DATE_KEY), 'start date'),
+    HabitatWorksEndDate: getDateString(request.yar.get(constants.redisKeys.LEGAL_AGREEMENT_END_DATE_KEY), 'end date'),
+
     ...geospatialOrLandBoundaryContext(request)
   }
 }
+const getFileNameByType = (files, desiredType) => {
+  const file = files.find(file => file.fileType === desiredType)
+  return file.fileName
+}
```

Comment:
Use of this function in check-and-submit.html results in display of [object Object] for most upload types. Please perform some refactoring here or elsewhere in this pull request to ensure file names are displayed.

Reply:
this was fixed already on friday